services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -221,21 +222,20 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)(impl
   ): AsyncResult[PipelineVersion] =
     getLastProjectVersion(project).map(projectVersion => getNewProjectVersion(projectVersion, optionUserVersion))
 
-  private def getLastProjectVersion(project: Project): Future[Option[PipelineVersion]] =
+  private def getLastProjectVersion(project: Project): Future[PipelineVersion] =
     getProjectVersions(project).flatMap {
       case Left(error)        => Future.failed(error)
-      case Right(List(_))     => Future.successful(None) // fixme why ignore the only version?
-      case Right(List(v, _*)) => Future.successful(Some(v))
-      // fixme how is this possible? Project must have at least one version project.version
-      case Right(List()) => Future.successful(Some(defaultProjectVersion))
+      case Right(List(v, _*)) => Future.successful(v)
+      case Right(List())      => Future.successful(project.version)
     }
 
   private def getNewProjectVersion(
-    optionProjectVersion: Option[PipelineVersion],
+    optionProjectVersion: PipelineVersion,
     optionUserVersion: Option[PipelineVersion]
   ): Either[VersioningException, PipelineVersion] =
     (optionProjectVersion, optionUserVersion) match {
-      case (Some(projectVersion), Some(userVersion)) =>
+      case (projectVersion, None) => Right(projectVersion.increaseRevision)
+      case (projectVersion, Some(userVersion)) =>
```

Comment:
Pattern matching on tuple `(PipelineVersion, Option[PipelineVersion])` makes no sense. Why do you need to pattern match on `projectVersion` if you don't do anything with it? Can't you just use it directly?

Reply:
Fixed.